buku

```diff
@@ -1492,132 +1545,131 @@ class BukuDb:
             q0 += ')'
 
         try:
-            return self._fetch(q0)
+            return self._fetch(q0 + f' ORDER BY {self._order(order)}')
         except sqlite3.OperationalError as e:
             LOGERR(e)
             return []
 
+    def _search_tokens(self, keyword: str, deep=False, regex=False, markers=False):
+        """Converts a keyword into a list of tokens, based on search parameters.
+        A token is a varied-length tuple of following values: (SQL field, deep, *SQL params)."""
+        deep = not regex and deep
+        if not markers or (re.sub(r'^\*', '', keyword) and not re.match(r'^[.:>#]', keyword)):
+            s = (keyword if not markers else re.sub(r'^\*', '', keyword))
+            if not s:
+                return []
+            tags = ([s] if regex and not markers else taglist(s.split(DELIM)))
+            return [('metadata', deep, s), ('url', deep, s), ('desc', deep, s)] + (tags and [('tags', deep, *tags)])
```

Comment:
Each token includes the field name, its own `deep` value (to account for special tags behaviour), and one or more parameter (again, to account for tags since they cannot include `,`).

Note that for a token to be matched, _all_ of its parameters must be matched (i.e. when searching for `foo,bar,baz` in tags, all three tags must be matched for the token to match).

And for the keyword to match, _any_ of its tokens must be matched.